src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.CreateDefaultHandler.Android.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Net.Http
+{
+    public partial class HttpClient
+    {
+        private const string CustomHandlerTypeEnvVar = "XA_HTTP_CLIENT_HANDLER_TYPE";
+
+        private static Type? s_customHandlerType;
+
+        private static HttpMessageHandler CreateDefaultHandler()
+        {
+            if (s_customHandlerType != null)
+            {
+#pragma warning disable IL2057
+                return (HttpMessageHandler)Activator.CreateInstance(s_customHandlerType)!;
+#pragma warning restore IL2057
+            }
+
+            string? envVar = Environment.GetEnvironmentVariable(CustomHandlerTypeEnvVar)?.Trim();
+            if (!string.IsNullOrEmpty(envVar))
+            {
+#pragma warning disable IL2072
```

Comment:
This pragma does not help, you need a proper solution or the type will be trimmed

Reply:
the type won't be trimmed - XA has an ILink sub-step to always preserve whatever defined in that env.var.

Reply:
Which can be easily disabled or won't run during development